client/sections-middleware.js

```diff
@@ -84,11 +84,11 @@ function createPageDefinition( path, sectionDefinition ) {
 			} )
 			.catch( error => {
 				console.error( error ); // eslint-disable-line
-				if ( ! LoadingError.isRetry() ) {
+				if ( ! LoadingError.isRetry() && process.env.NODE_ENV !== 'development' ) {
 					LoadingError.retry( sectionDefinition.name );
 				} else {
 					dispatch( { type: 'SECTION_SET', isLoading: false } );
-					LoadingError.show( sectionDefinition.name );
+					LoadingError.show( context, sectionDefinition.name );
```

Comment:
`context` definitely has to be passed in here, you're right. Good catch!

However, I don't see [`LoadingErrorMessage`](https://github.com/Automattic/wp-calypso/blob/e80201fc18ee8c6e4c5c9a63b5c43d2ad83a4a8a/client/layout/error.jsx#L48-L52); this will need something additional to work. I'm just getting an empty page.

I tested adding `next()` here but no dice.

I also tested adding this:
```js
import { makeLayout, render as clientRender } from 'controller';
// ...
LoadingError.show( context, sectionDefinition.name );
makeLayout( context, noop );
clientRender( context );
```

But I'm getting:
>warning.js:33 Warning: Can only update a mounted or mounting component. This usually means you called setState, replaceState, or forceUpdate on an unmounted component. This is a no-op.
>
>Please check the code for the AsyncLoad component.

— I'll have another look with fresh eyes tomorrow.

Reply:
Ah — noticed only now that you listed getting the error visible as a _follow up_. Makes sense! :thumbsup: